sklearn/feature_extraction/tests/test_text.py

```diff
@@ -328,6 +328,31 @@ def test_fit_countvectorizer_twice():
     assert_not_equal(X1.shape[1], X2.shape[1])
 
 
+def test_countvectorizer_custom_token_pattern():
+    corpus = [
+        'This is the first document.',
+        'This document is the second document.',
+        'And this is the third one.',
+        'Is this the first document?',
+    ]
+    # With no capturing group
```

Comment:
I don't think we need to test the no capturing group case, really, as that is already tested.

But we do need a test for the 1 capturing group case.